addons/interaction/functions/fnc_passThrowable.sqf

```diff
@@ -0,0 +1,36 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Timi007
+ * Pass throwable to another unit.
+ *
+ * Arguments:
+ * 0: Unit to pass the throwable to <OBJECT>
+ * 1: Unit that passes the throwable <OBJECT>
+ * 2: Throwable classname <STRING>
+ * 3: Play passing animation <BOOL> (default: true)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_target, _player, "HandGrenade"] call ace_interaction_fnc_passThrowable
+ *
+ * Public: No
+ */
+
+params [["_target", objNull, [objNull]], ["_player", objNull, [objNull]], ["_throwable", "", [""]], ["_animate", true, [true]]];
+TRACE_4("Pass throwable params",_player,_target,_throwable,_animate);
+
+if (isNull _target) exitWith {ERROR("Target is null.")};
+if (isNull _player) exitWith {ERROR("Player is null.")};
+if (_throwable isEqualTo "") exitWith {ERROR("No throwable specified.")};
+if !([_target, _throwable] call CBA_fnc_canAddItem) exitWith {ERROR("Cannot add throwable to target due to lack of inventory space.")};
+
+if (_animate) then {[_player, "PutDown"] call EFUNC(common,doGesture)};
+
+_player removeItem _throwable;
```

Comment:
The condition is evaluated only once a second, which allows up to 1 second where you could have no grenades but still be able to pass a grenade.
The fix is to check if the item was successfully removed and stop before doing anything else:

```suggestion
if !([_player, _throwable] call CBA_fnc_removeItem) exitWith {};

if (_animate) then {[_player, "PutDown"] call EFUNC(common,doGesture)};
```